src/main/java/wedlog/address/logic/commands/GuestEditCommand.java

```diff
@@ -0,0 +1,329 @@
+            return false;
+        }
+
+        GuestEditCommand otherGuestEditCommand = (GuestEditCommand) other;
+        return index.equals(otherGuestEditCommand.index)
+                && editGuestDescriptor.equals(otherGuestEditCommand.editGuestDescriptor);
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
+                .add("index", index)
+                .add("editGuestDescriptor", editGuestDescriptor)
+                .toString();
+    }
+
+    /**
+     * Stores the details to edit the guest with.
+     * For non-empty fields, the field value will replace the corresponding field value of the existing guest.
+     * For empty fields, the existing field value will be deleted.
+     */
+    public static class EditGuestDescriptor {
+        private Name name;
+        private Phone phone;
+        private Email email;
+        private Address address;
+        private RsvpStatus rsvp;
+        private DietaryRequirements dietary;
+        private TableNumber table;
+        private Set<Tag> tags;
+        private boolean isNameEdited = false;
+        private boolean isPhoneEdited = false;
+        private boolean isEmailEdited = false;
+        private boolean isAddressEdited = false;
+        private boolean isRsvpEdited = false;
+        private boolean isDietaryEdited = false;
+        private boolean isTableEdited = false;
+        private boolean isTagsEdited = false;
+
+        public EditGuestDescriptor() {}
+
+        /**
+         * Copy constructor.
+         * A defensive copy of {@code tags} is used internally.
+         */
+        public EditGuestDescriptor(EditGuestDescriptor toCopy) {
+            name = toCopy.name;
+            phone = toCopy.phone;
+            email = toCopy.email;
+            address = toCopy.address;
+            rsvp = toCopy.rsvp;
+            dietary = toCopy.dietary;
+            table = toCopy.table;
+            tags = (toCopy.tags != null) ? new HashSet<>(toCopy.tags) : null;
+
+            isNameEdited = toCopy.isNameEdited;
+            isPhoneEdited = toCopy.isPhoneEdited;
+            isEmailEdited = toCopy.isEmailEdited;
+            isAddressEdited = toCopy.isAddressEdited;
+            isRsvpEdited = toCopy.isRsvpEdited;
+            isDietaryEdited = toCopy.isDietaryEdited;
+            isTableEdited = toCopy.isTableEdited;
+            isTagsEdited = toCopy.isTagsEdited;
+        }
+
+        /**
+         * Returns true if at least one field is edited.
+         */
+        public boolean isAnyFieldEdited() {
+            return CollectionUtil.isAnyTrue(isNameEdited, isPhoneEdited, isEmailEdited, isAddressEdited, isRsvpEdited,
+                    isDietaryEdited, isTableEdited, isTagsEdited);
+        }
+
+        public void setName(Name name) {
+            isNameEdited = true;
+            this.name = name;
+        }
+
+        public Optional<Name> getName() {
+            return Optional.ofNullable(name);
+        }
```

Comment:
Just a nit, but these methods no longer need to return `Optional`.
```suggestion
        public Name getName() {
            return name;
        }
```